### DISCUSSION TITLE
Exporting metrics to MongoDB from NetData Cloud

### DISCUSSION THREAD

**Discussion by isuchy** [netdata/netdata#15481] [Q&A]
Hi guys,

I'm facing to the problem similar to this [one](https://github.com/netdata/netdata/issues/9280) with the difference that I monitor processes in **Google Colab** environment. 

I installed NetData according to the [blogpost](https://blog.netdata.cloud/google-colab-monitoring/) and it works fine. I need to send data to the MongoDB for further analysis. I set up `exporting.conf` file as following:

```
[exporting:global]
    enabled = yes
    # send configured labels = yes
    # send automatic labels = no
    update every = 10

[mongodb:my_mongodb_instance]
    enabled = yes
    destination = mongodb+srv://user:password@cluster0.xxxxxxx.mongodb.net/
    database = netdata
    collection = apps
    data source = average
    # prefix = netdata
    # hostname = my_hostname
    # update every = 10
    # buffer on failures = 10
    # timeout ms = 20000
    # send names instead of ids = yes
    send charts matching = apps.*
    # send hosts matching = localhost *

```

As it was recommended in thread mentioned above, I also installed the libmongoc package before NetData installation as described [here](http://mongoc.org/libmongoc/current/installing.html). When I check the `error.log`, still the same result

![image](https://github.com/netdata/netdata/assets/19557763/56c3d6ac-aacc-435e-9e74-d912783e2295)

What else should I do to make it work? 
Thank you in advance for any guidance.

**Comment by andrewm4894** [comment]
I think you would need to follow steps here: https://learn.netdata.cloud/docs/exporting/mongodb#prerequisites

1. install libmongoc http://mongoc.org/libmongoc/current/installing.html
2. then install netdata from source where it should pickup fact that libmongoc is available: https://learn.netdata.cloud/docs/installing/build-the-netdata-agent-yourself/compile-from-source-code

Then your configs etc will work

This might all get a bit annoying to have to do in colab all the time though (if google colab even lets you do this level of installing stuff - not sure myself). 

You might be better to have the netdata on your colabs stream to a netdata parent and then this parent could export to mongo if you want and that way you'd only need to install and manage the exporting on the netdata parent and when installing netdata on your colab you just configure it to stream to the netdata parent. I think something like that could be easier if you end up having to use different colab instances frequently say and just want a quicker sort of install script for setting them up.

**Comment by isuchy** [comment]
Hi @andrewm4894,

thanks for your reply. As stated in prerequisites paragraph and also 2nd point of your suggestion, it seems to be necessary to make a build from source code. I'd like to avoid this step, especially in Google Colab env, it's a pain due to plenty of limitations. But I understand, there is no other way how to get working Exporting plugin.

Regarding your other recommendation, it's too complicated, since I only need functional NetData monitoring for some experiments in Colab, not for frequent or even regular use.

Anyway, I'm going to try to compile NetData and when it works I'll close the issue as an accepted solution.

Thanks.

**Reply by andrewm4894** [reply]
if its about shipping the data off of the agent for each experiment you could maybe do similar by using the agent rest api to get what you want and then just dump that somewhere like s3 bucket etc. 

https://learn.netdata.cloud/api#/data/dataQuery2

Or if using python maybe something like https://github.com/netdata/netdata-pandas to get what data you want from the agent into a df and save it where ever you want from there.

**Comment by isuchy** [comment]
Hello @andrewm4894,

I have decided to use MongoDB, because I found it easier to implement than agent REST API particularly collecting data from output. 

Anyway I have several issues with implementation in Google Colab environment. Do you think that the platform is supported, or someone from team has performed any test? Unfortunately I'm not a Linux guy but I can see there are plenty of restrictions about calling system services. As far as I know, each Jupyter Notebook runs in container.

I have followed the instructions and it looks like this (I also do emphasize the broken parts)


`!apt-get update`

`!curl -Ss 'https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/install-required-packages.sh' >/tmp/install-required-packages.sh && bash /tmp/install-required-packages.sh -i netdata-all -y`

`!apt-get install libmongoc-dev zlib1g-dev uuid-dev libuv1-dev liblz4-dev libssl-dev libelf-dev libmnl-dev libjson-c-dev libprotobuf-dev protobuf-compiler gcc g++ make git autoconf autoconf-archive autogen automake pkg-config curl cmake`

**##### START OF CRASHED PART ######################################################################**
```
!git clone https://github.com/netdata/netdata.git --depth=100 --recursive
%cd /content/netdata
!./netdata-installer.sh --dont-wait --enable-exporting-mongodb
```

![image](https://github.com/netdata/netdata/assets/19557763/9f2d80c6-b29f-4bea-be7b-f3a01f4e119e)
![image](https://github.com/netdata/netdata/assets/19557763/f38252fa-3c74-49a4-a42b-a1e732079dbd)
![image](https://github.com/netdata/netdata/assets/19557763/0d538441-3628-46fb-8814-829e5d682794)
**###### END OF CRASHED PART ######################################################################**

`!wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh --stable-channel --claim-token TOKEN_REPLACED --claim-rooms b3dc2c32-0eb5-4d74-a3fc-d850215a4028 --claim-url https://app.netdata.cloud`

**##### START OF CRASHED PART #####################################################################**
`!/etc/netdata/edit-config exporting.conf`

![image](https://github.com/netdata/netdata/assets/19557763/1928abd5-d53a-40d3-8fbd-13867d047923)
**###### END OF CRASHED PART ######################################################################**

**##### START OF CRASHED PART #####################################################################**
`!netdata `

![image](https://github.com/netdata/netdata/assets/19557763/78318a49-6519-460e-a07a-8e2bd0bd4b83)
**###### END OF CRASHED PART ######################################################################**


Could you please to try out, if it works for you? I've included the Jupyter notebook for faster checkout, so you don't have to type it all out. 

Thanks a lot again.

[Netdata.ipynb.zip](https://github.com/netdata/netdata/files/12136741/Netdata.ipynb.zip)
